usr/src/boot/efi/libefi/efi_console.c

```diff
@@ -457,8 +457,13 @@ efi_framebuffer_setup(void)
 	}
 	shadow_sz = EFI_SIZE_TO_PAGES(efifb.fb_width * efifb.fb_height *
 	    sizeof (*shadow_fb));
+#if defined(__aarch64__)
+	status = BS->AllocatePages(AllocateAnyPages, EfiLoaderData,
```

Comment:
eventually, we're going to want to hide the allocation differences better, I think.

Reply:
Yeah, Toomas was saying there are reasons for i86pc to be this way, but it gets pretty ugly to conditionalise each one.